bears/general/IndentationBear.py

```diff
@@ -0,0 +1,209 @@
+from coalib.bears.LocalBear import LocalBear
+from bears.general.AnnotationBear import AnnotationBear
+from bears.general.AnnotationBear import starts_within_ranges
+from coalib.parsing.StringProcessing.Core import unescaped_search_for
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.results.SourceRange import SourceRange
+from coalib.results.Result import Result, RESULT_SEVERITY
+from coalib.results.AbsolutePosition import AbsolutePosition
+from coalib.results.Diff import Diff
+
+
+class IndentationBear(LocalBear):
+
+    def run(self,
+            filename,
+            file,
+            dependency_results: dict,
+            language: str,
+            language_family: str,
+            use_spaces: bool=True,
+            tab_width: int=4):
+        """
+        It is a generic indent bear, which looks for a start and end
+        indent eg: { : }  if they are specified.
+        eg: "{"  where { is an
+        indent start.
+
+        It does not however support hanging indents or absolute indents of
+        any sort, also indents based on keywords are not supported yet.
+        for example:
+
+            if(something)
+            does not get indented
+
+        undergoes no change.
+
+        :param language_family:
+            Language family to be used for indentation.
+        :param language:
+            Language to be used for indentation.
+        :param use_spaces:
+            Insert tabs instead of spaces for indentation.
+        :param tab_width:
+            No. of spaces to insert for indentation.
+            Only Applicable if use_tabs_instead_of_spaces is False
+        """
+        lang_settings_dict = LanguageDefinition(language, language_family)
+        annotation_dict = dependency_results[AnnotationBear.name][0].contents
+        indent_types = dict(lang_settings_dict["indent_types"])
+        indent_levels = self.get_indent_levels(
+                                file, filename, indent_types, annotation_dict)
+
+        # This happens only in case of unmatched indents.
+        if type(indent_levels) is type(''):
```

Comment:
type comparision is something else FWIW @Makman2 !

Sincerely,

Lasse Schuirmann

lasse@schuirmann.net
http://coala-analyzer.org/ - http://viperdev.io/ - http://gitmate.com/

On 29 May 2016 at 22:38, Mischa Krüger <notifications@github.com> wrote:

> In bears/general/IndentationBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/468#discussion_r65011028>
> :
>
> > +        :param language:
> > +            Language to be used for indentation.
> > +        :param use_spaces:
> > +            Insert tabs instead of spaces for indentation.
> > +        :param tab_width:
> > +            No. of spaces to insert for indentation.
> > +            Only Applicable if use_tabs_instead_of_spaces is False
> > +        """
> > +        lang_settings_dict = LanguageDefinition(language, language_family)
> > +        annotation_dict = dependency_results[AnnotationBear.name][0].contents
> > +        indent_types = dict(lang_settings_dict["indent_types"])
> > +        indent_levels = self.get_indent_levels(
> > +                                file, filename, indent_types, annotation_dict)
> > +
> > +        # This happens only in case of unmatched indents.
> > +        if type(indent_levels) is type(''):
>
> why not using instanceof?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/468/files/6eb3e1fbd09861e594a88bda408b50cc9cf167d8#r65011028>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFc6KMjvgpF6fZFIHAyq_vJTNR-4RKElks5qGflUgaJpZM4IpJbD>
> .
>